pallets/qa-tools/src/lib.rs

```diff
@@ -82,6 +84,13 @@ pub mod pallet {
     pub type WhitelistedCallers<T: Config> =
         StorageValue<_, BoundedBTreeSet<T::AccountId, T::QaToolsWhitelistCapacity>>;
 
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        AddedToWhitelist { user: T::AccountId },
+        RemovedFromWhitelist { user: T::AccountId },
+    }
+
```

Comment:
You don't need to introduce events because I will delete the whitelist in my PR

Reply:
I've added them before I found this out; will remove 👍